--- a/test/MaterializedViewsTest.cpp
+++ b/test/MaterializedViewsTest.cpp
@@ -743,7 +743,7 @@
   {
     qlv().writeMaterializedView("testView7", simpleWriteQuery_);
     auto view = std::make_shared<MaterializedView>(testIndexBase_, "testView7");
-    view->parsedQuery_ = std::nullopt;
+    view->xpodTestParsedQuery() = std::nullopt;
     materializedViewsQueryAnalysis::QueryPatternCache c;
     EXPECT_FALSE(c.analyzeView(view));
   }
@@ -786,8 +786,8 @@
         std::make_shared<ad_utility::CancellationHandle<>>();
     static constexpr size_t dummyTimeLimit = 1000 * 60 * 60;  // 1 hour
     std::chrono::milliseconds timeLimit{dummyTimeLimit};
-    server.writeMaterializedView("testViewFromServer", query, requestTimer,
-                                 cancellationHandle, timeLimit);
+    server.xpodTestWriteMaterializedView("testViewFromServer", query, requestTimer,
+                                         cancellationHandle, timeLimit);
   }
 
   // Test the preloading of materialized views on server start.
@@ -796,7 +796,7 @@
     config.preloadMaterializedViews_ = {"testViewForServerPreload"};
     qlv().writeMaterializedView("testViewForServerPreload", simpleWriteQuery_);
     Server server{4321, 1, "accessToken", config};
-    EXPECT_TRUE(server.qlever_.materializedViewsManager()->isViewLoaded(
+    EXPECT_TRUE(server.xpodTestQlever().materializedViewsManager()->isViewLoaded(
         "testViewForServerPreload"));
   }
 
