--- a/src/engine/Values.cpp
+++ b/src/engine/Values.cpp
@@ -12,6 +12,7 @@
 #include "engine/CallFixedSize.h"
 #include "util/Exception.h"
 #include "util/HashSet.h"
+#include "XpodQleverPhysicalValuesContextBridge.hpp"
 
 // ____________________________________________________________________________
 Values::Values(QueryExecutionContext* qec, SparqlValues parsedValues)
@@ -133,7 +134,10 @@ void Values::writeValues(IdTable* idTablePtr, LocalVocab* localVocab) {
       const TripleComponent& tc = row[colIdx];
       // TODO<joka921> We don't want to move, but also don't want to
       // unconditionally copy.
-      Id id = TripleComponent{tc}.toValueId(getIndex(), *localVocab);
+      std::optional<Id> physicalId =
+          xpod::qlever::physicalValuesIdFromContext(*getExecutionContext(), tc);
+      Id id = physicalId.value_or(
+          TripleComponent{tc}.toValueId(getIndex(), *localVocab));
       idTable(rowIdx, colIdx) = id;
       if (id.getDatatype() == Datatype::LocalVocabIndex) {
         ++numLocalVocabPerColumn[colIdx];
