diff --git a/src/engine/OptionalJoin.cpp b/src/engine/OptionalJoin.cpp
--- a/src/engine/OptionalJoin.cpp
+++ b/src/engine/OptionalJoin.cpp
@@ -4,6 +4,7 @@
 //         Florian Kramer (florian.kramer@netpun.uni-freiburg.de)
 
 #include "engine/OptionalJoin.h"
+#include "XpodQleverPhysicalIndexScanContextBridge.hpp"
 
 #include "engine/AddCombinedRowToTable.h"
 #include "engine/CallFixedSize.h"
@@ -144,6 +145,7 @@ Result OptionalJoin::computeResult(bool requestLaziness) {
   const bool isTwoColumnSpecialOptionalJoin =
       implementation_ == Implementation::OnlyUndefInLastJoinColumnOfLeft &&
       _joinColumns.size() == 2;
-  if (getRuntimeParameter<&RuntimeParameters::prefilteredOptionalJoin_>() &&
+  if (xpod::qlever::physicalIndexFromContext(*getExecutionContext()) == nullptr &&
+      getRuntimeParameter<&RuntimeParameters::prefilteredOptionalJoin_>() &&
       (_joinColumns.size() == 1 || isTwoColumnSpecialOptionalJoin)) {
     if (auto indexScan =
             std::dynamic_pointer_cast<IndexScan>(_right->getRootOperation())) {
