diff --git a/src/engine/QueryPlanner.cpp b/src/engine/QueryPlanner.cpp
--- a/src/engine/QueryPlanner.cpp
+++ b/src/engine/QueryPlanner.cpp
@@ -2948,5 +2948,12 @@ qlever::index::GraphFilter<TripleComponent> QueryPlanner::getActiveGraphs()
     return Filter::Blacklist(TripleComponent{
         ad_utility::triple_component::Iri::fromIriref(DEFAULT_GRAPH_IRI)});
   }
+  if (_qec != nullptr &&
+      xpod::qlever::physicalIndexFromContext(*_qec) != nullptr) {
+    ad_utility::HashSet<TripleComponent> defaultGraphs;
+    defaultGraphs.emplace(TripleComponent{
+        ad_utility::triple_component::Iri::fromIriref(DEFAULT_GRAPH_IRI)});
+    return Filter::Whitelist(std::move(defaultGraphs));
+  }
   return Filter::All();
 }
